libbeat/cfgfile/reload.go

```diff
@@ -15,3 +31,158 @@ type ReloadConfig struct {
 	Period  time.Duration `config:"period"`
 	Enabled bool          `config:"enabled"`
 }
+
+type Wrapper interface {
```

Comment:
This looks like the interface to a factory. Maybe you could call it `RunnerFactory` with a `Create` method. 

I think it would be better to integrate the `uint64` value into the `Runner` that is returned. I'm assuming this is some sort of identifier? So probably an `ID()` method in `Runner` would make sense.